packages/dds/tree/src/simple-tree/discrepancies.ts

```diff
@@ -294,21 +443,42 @@ function* getFieldDiscrepancies(
 		} satisfies AllowedTypeDiscrepancy;
 	}
 
-	if (view.kind !== stored.kind) {
+	const viewKind =
+		convertFieldKind.get(view.kind) ??
+		fail("A conversion from a FieldKind to a FlexFieldKind should exist");
+
+	// This checks if the field kind in the view schema is not compatible with the stored schema.
+	// We cannot detect if the view schema is a sequence using the kind property so it is passed in separately.
+	// TODO: This is a temporary workaround until the comparison logic is redesigned.
+	if (
+		(viewKindIsSequence && stored.kind !== "Sequence") ||
+		(!viewKindIsSequence && viewKind.identifier !== stored.kind)
+	) {
 		yield {
 			identifier,
 			fieldKey,
 			mismatch: "fieldKind",
-			view: view.kind,
+			view: viewKind.identifier,
 			stored: stored.kind,
 		} satisfies FieldKindDiscrepancy;
 	}
 }
 
+/**
+ * Tracks discrepancies between a view schema and a stored schema for nodes that are treated as object nodes in the stored schema.
+ * This includes both view object nodes and view array nodes.
+ *
+ * This function yields discrepancies in the following cases:
+ * 1. If the view schema has fields that are not present in the stored schema.
+ * 2. If the stored schema has fields that are not present in the view schema.
+ * 3. If the field kind or allowed types of a field in the view schema is not compatible with the stored schema.
+ * - This function recurses into the common fields of the view and stored schemas to find discrepancies using {@link getFieldDiscrepancies}.
```

Comment:
I think you are wrong about this recursing. If it recursed, then it would not handle recursive types (as it would stack overflow instead). Also, your formatting is a bit messed up.

```suggestion
 * @remarks
 * This function yields discrepancies in the following cases:
 *
 * 1. If the view schema has fields that are not present in the stored schema.
 * 2. If the stored schema has fields that are not present in the view schema.
 * 3. If the field kind or allowed types of a field in the view schema is not compatible with the stored schema.
 *
 * This function includes discrepancies within the common fields and their allowed types, but does NOT recurse to report and discrepancies within the node types referenced by those fields.
```